### DISCUSSION TITLE
Escape Hyphens in Function Names for AppSettings

### DISCUSSION THREAD

**Discussion by davidmrdavid** [Azure/Azure-Functions#1944] [Design discussions]
Discussion for: https://github.com/Azure/Azure-Functions/issues/1942

**Comment by brettsam** [comment]
Questions from discussion:
- Is the proposal that the Portal always escapes? Does the Portal need to know that it can only do this in v4?
  - Portal could try/catch around setting the value. If it fails, retry with an escaped function name.
- What if, instead, we had a single key like `DisabledFunctions` -- with a semi-colon disabled list of disabled Functions?
  - Would need to support both models in v3. And determine what happens if there is conflicting detail between the two?
  - However, This would be adding yet another way to disable functions, which would be hard to manage.
- Components impacted should include Portal (or possibly the management API operation). If this is all behind an API, then checking version could be simpler.
- Does ScaleController need work here to support enable/disable?
- We should be able to backport this to v3 to simplify the matrix of support. 
- Need to define the escape char in Linux. This requires some research. @davidmrdavid to investigate.

Approving this -- but finding the escape sequence that works on Windows and Linux is still required.

This does not constitute a breaking change, so we can remove this from the breaking change list and fix it in v3 and v4 whenever we want.
